File: mandlebrot.js

const iSize     = 2048;
const iMaxIters = 512;
const fBailout  = 4.0;
const fYMin     = -1.25;
const fYMax     = 1.25;
const fYTot     = fYMax - fYMin;
const fXMax     = 0.5;
const fXMin     = fXMax - fYTot;
const fXTot     = fXMax - fXMin;
const fStep     = fYTot / iSize;
let iCnt      = 0;
let fCY       = fYMax;

for (let y = 0; y < iSize/2; y++, fCY -= fStep) {
  let fCX = fXMin;
  for (let x = 0; x < iSize; x++, fCX += fStep) {
    let fZX = fCX;
    let fZY = fCY;
    let fTest;
    let i   = iMaxIters;
    
    do {
        let fZX2   = fZX * fZX;
        let fZY2   = fZY * fZY;
      fTest  = fZX2 + fZY2;
      let fNewZX = fZX2 - fZY2 + fCX;
      let fNewZY = 2.0 * fZX * fZY + fCY;
      fZY    = fNewZY;
      fZX    = fNewZX;
    } while (--i && fTest < fBailout);
    iCnt += i ? 0 : 1;
  }
}

let fArea = 2 * iCnt * (fXTot * fYTot) / (iSize * iSize );

console.log( "Done: " + fArea);

File: mandlebrot.c
#include <stdio.h>

int main() {
  int iSize       = 2048;
  int iMaxIters   = 512;
  double fBailout  = 4.0;
  double fYMin     = -1.25;
  double fYMax     = 1.25;
  double fYTot     = fYMax - fYMin;
  double fXMax     = 0.5;
  double fXMin     = fXMax - fYTot;
  double fXTot     = fXMax - fXMin;
  double fStep     = fYTot / iSize;
  int iCnt        = 0;
  double fCY       = fYMax;
  for (int y = 0; y < iSize/2; y++, fCY -= fStep) {
    double fCX = fXMin;
    for (int x = 0; x < iSize; x++, fCX += fStep) {
      double fZX = fCX;
      double fZY = fCY;
      double fTest;
      int i   = iMaxIters;
      
      do {
        double fZX2   = fZX * fZX;
        double fZY2   = fZY * fZY;
        fTest  = fZX2 + fZY2;
        double fNewZX = fZX2 - fZY2 + fCX;
        double fNewZY = 2.0 * fZX * fZY + fCY;
        fZY    = fNewZY;
        fZX    = fNewZX;
      } while (--i && fTest < fBailout);
      iCnt += i ? 0 : 1;
    }
  }

  double fArea = 2 * iCnt * (fXTot * fYTot) / (iSize * iSize );

  printf( "Done: %f", fArea);
  return 0;
}
